clientconn.go

```diff
@@ -584,6 +601,12 @@ func (csm *connectivityStateManager) getNotifyChan() <-chan struct{} {
 	return csm.notifyChan
 }
 
+func (csm *connectivityStateManager) close() {
+	csm.mu.Lock()
+	defer csm.mu.Unlock()
```

Comment:
We don't need to hold the lock here since the `pubSub` field is initialized at creation time and is never written to after that.